fieldservice_fleet/models/fleet_vehicle.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (C) 2019 Open Source Integrators
+# Copyright (C) 2019 Brian McMaster
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class FleetVehicle(models.Model):
+    _inherit = 'fleet.vehicle'
+
+    is_fsm_vehicle = fields.Boolean(
+        string='Is used for Field Service?'
+    )
+
+    @api.onchange('is_fsm_vehicle')
+    def onchange_is_fsm_vehicle(self):
+        if is_fsm_vehicle == True:
+            # Create fsm.vehicle linked to fleet.vehicle
+            fsm_vehicle = self.env['fsm.vehicle'].create(
+                {'fleet_vehicle_id': self.id}
+            )
```

Comment:
@max3903 
Can you help me here?

I am receiving error

```
The operation cannot be completed:
- Create/update: a mandatory field is not set.
- Delete: another model requires the record being deleted. If possible, archive it instead.

Model: Vehicle (fleet.vehicle), Field: Model (model_id)
```
It is trying to create another fleet.vehicle and I am not setting the required fields for fleet.vehicle, causing the error.  However I don't want to create a new fleet.vehicle, but simply link it to a newly created fsm.vehicle

Reply:
@brian10048 This is the same requirement as in fieldservice (res.partner --> fsm.location). Check there how I implemented the conversion with a server action and a wizard...